### PR TITLE
Add tags support for activities, sub-orchestrations, and client APIs

### PR DESCRIPTION
**Pull request by torosent**
# Summary

## What changed?
- Added `tags` property (key-value metadata) to `OrchestrationState` for reading tags from orchestration state
- Added `StartOrchestrationOptions` interface with `tags` support for scheduling new orchestrations
- Added `tags` to `TaskOptions` for activities and `SubOrchestrationOptions` for sub-orchestrations
- Extended client `scheduleNewOrchestration` with an overload accepting `StartOrchestrationOptions`
- Extended proto helper functions `newScheduleTaskAction` and `newCreateSubOrchestrationAction` to accept tags  
- Added tag preservation during activity/sub-orchestration retry scheduling
- Created shared `mapToRecord` utility in `tags.util.ts` for converting proto tag maps to TypeScript records
- Added comprehensive unit tests for tag round-trip in orchestration state, activities, and sub-orchestrations
- Added DTS e2e test for tag persistence

## Why is this change needed?
- Tags provide a way to attach custom key-value metadata to orchestrations, activities, and sub-orchestrations
- This enables better organization, filtering, and tracking of workflow instances
- The proto already supports tags on `CreateInstanceRequest`, `OrchestrationState`, `ScheduleTaskAction`, `CreateSubOrchestrationAction`, and `SubOrchestrationInstanceCreatedEvent`

## Issues / work items
- N/A

---

# Project checklist
- [x] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `CHANGELOG.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:

---

# AI-assisted code disclosure (required)

## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot (Claude)
- AI-assisted areas/files:
  - `packages/durabletask-js/src/utils/tags.util.ts` (new file)
  - `packages/durabletask-js/src/task/options/task-options.ts`
  - `packages/durabletask-js/src/task/options/index.ts`
  - `packages/durabletask-js/src/index.ts`
  - `packages/durabletask-js/src/orchestration/orchestration-state.ts`
  - `packages/durabletask-js/src/orchestration/index.ts`
  - `packages/durabletask-js/src/client/client.ts`
  - `packages/durabletask-js/src/utils/pb-helper.util.ts`
  - `packages/durabletask-js/src/worker/runtime-orchestration-context.ts`
  - `packages/durabletask-js/test/query-apis.spec.ts`
  - `packages/durabletask-js/test/orchestration_executor.spec.ts`
  - `test/e2e-azuremanaged/orchestration.spec.ts`
- What you changed after AI output: Reviewed and refined tag handling, consolidated duplicate mapToRecord functions, strengthened test assertions

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing

## Automated tests
- Result: Passed (132 unit tests pass)

## Manual validation (only if runtime/behavior changed)
- Environment (OS, Node.js version, components): N/A
- Steps + observed results: N/A
- Evidence (optional):

---

# Notes for reviewers
- Tags are only supported in DTS (Durable Task Scheduler), not in local sidecar - the sidecar e2e tests were intentionally not added
- The `mapToRecord` utility handles undefined proto maps gracefully
- Tag preservation during retries ensures consistency when activities/sub-orchestrations fail and are rescheduled

### CODE REVIEW

File: test/e2e-azuremanaged/orchestration.spec.ts [hunk 1]
```diff
@@ -86,6 +86,23 @@ describe("Durable Task Scheduler (DTS) E2E Tests", () => {
...
+      return "done";
+    };
+
+    taskHubWorker.addOrchestrator(orchestrator);
+    await taskHubWorker.start();
+
+    const tags = { env: "test", owner: "durable" };
+    const id = await taskHubClient.scheduleNewOrchestration(orchestrator, undefined, { tags });
+    const state = await taskHubClient.waitForOrchestrationCompletion(id, undefined, 30);
+
```
Comment (YunchuWang) [inline test/e2e-azuremanaged/orchestration.spec.ts]: tests for scheduling activity and suborch with tags too? can be in this same testcase?
